### PR TITLE
Update readme generator to include information about InteractiveBrowserCredential and our support policy

### PR DESCRIPTION
**Pull request by JonathanCrd**
Solves:

- https://github.com/Azure/autorest.typescript/issues/1274
- https://github.com/Azure/autorest.typescript/issues/1306

This PR modify the readme template to include a line pointing to our support policy, and a comment inside an `@azure/identity` code snippet clarifying that for client-side applications running in the browser, `InteractiveBrowserCredential` should be user instead of `DefaultAzureCredential`.

Integration and smoke tests were regenerated as well.

### CODE REVIEW

File: src/generators/static/README.md.hbs [hunk 1]
```diff
@@ -64,13 +66,15 @@ For more information about how to create an Azure AD Application check out [this
 ```javascript
 const { {{ clientClassName }} } = require("{{ clientPackageName }}");
 const { DefaultAzureCredential } = require("@azure/identity");
+// For client-side applications running in the browser, use InteractiveBrowserCredential instead of DefaultAzureCredential. See https://aka.ms/azsdk/js/identity/examples for more details.
```
Comment (joheredi) [inline src/generators/static/README.md.hbs:69]: maybe we can add a commented out example of how to use the interactiveBrowserCredential?

```typescript
 // const credential = new InteractiveBrowserCredential({
 //   tenantId: "<YOUR_TENANT_ID>",
 //   clientId: "<YOUR_CLIENT_ID>"
 // });
// const client = new {{ clientClassName }}(credential, subscriptionId);
```

**Review by joheredi — APPROVED** [review]
Looks good! I just wonder if we can add a sample of how to use the interactiveBrowserCredential?

File: src/generators/static/README.md.hbs (hunk 1 above)
Comment (JonathanCrd) [reply src/generators/static/README.md.hbs:69]: I like your idea, let me add the example :)

**Review by sarangan12 — APPROVED** [review]
LGTM

**Pull request merged**